packages/next/src/client/components/react-dev-overlay/shared.ts

```diff
@@ -232,8 +232,13 @@ export function useErrorOverlayReducer(
         getOwnerStack(event.error) !== getOwnerStack(pendingEvent.error)
       )
     })
-    pendingEvents.push(pendingEvent)
-    return pendingEvents
+    // If there's nothing filtered out, the event is a brand new error
+    if (pendingEvents.length === events.length) {
+      pendingEvents.push(pendingEvent)
+      return pendingEvents
+    }
+    // Otherwise remain the same events
+    return events
```

Comment:
Shouldn't this still be deduped?

```suggestion
    // Otherwise remain the same events
    return pendingEvents
```

Reply:
`pendingEvents` is returned from `events.filter` which is a new instance. This fix is intended to return the original `events`